tests/unit/model/designation-test.js

```diff
@@ -54,3 +54,191 @@ test('Relationships with other models', function(assert){
+    store.createRecord('orders_package', {id: 2, state: 'designated'});
+    store.createRecord('orders_package', {id: 3, state: 'dispatched'});
+
+    orders_package1 = store.peekRecord('orders_package',1);
+    orders_package2 = store.peekRecord('orders_package',2);
+    orders_package3 = store.peekRecord('orders_package',3);
+
+    model.get('ordersPackages').pushObject(orders_package1);
+    model.get('ordersPackages').pushObject(orders_package2);
+    model.get('ordersPackages').pushObject(orders_package3);
+  });
+
+  designatedOrdersPackagesIds = model.get('designatedOrdersPackages').getEach('id');
+
+  assert.equal(designatedOrdersPackagesIds.get('length'),2);
+  assert.equal(designatedOrdersPackagesIds[0], orders_package1.get('id'));
+  assert.equal(designatedOrdersPackagesIds[1], orders_package2.get('id'));
+});
+
+test('check dispatchedOrdersPackages returns only dispatched orders_packages', function(assert){
+  assert.expect(3);
+  const model = this.subject();
+  var store = this.store();
+  var orders_package1 = null;
+  var orders_package2 = null;
+  var orders_package3 = null;
+  var dispatchedOrdersPackagesIds = null;
+
+  Ember.run(function(){
+    store.createRecord('orders_package', {id: 1, state: 'dispatched'});
+    store.createRecord('orders_package', {id: 2, state: 'dispatched'});
+    store.createRecord('orders_package', {id: 3, state: 'designated'});
+
+    orders_package1 = store.peekRecord('orders_package',1);
+    orders_package2 = store.peekRecord('orders_package',2);
+    orders_package3 = store.peekRecord('orders_package',3);
+
+    model.get('ordersPackages').pushObject(orders_package1);
+    model.get('ordersPackages').pushObject(orders_package2);
+    model.get('ordersPackages').pushObject(orders_package3);
+  });
+
+  dispatchedOrdersPackagesIds = model.get('dispatchedOrdersPackages').getEach('id');
+
+  assert.equal(dispatchedOrdersPackagesIds.get('length'),2);
+  assert.equal(dispatchedOrdersPackagesIds[0], orders_package1.get('id'));
+  assert.equal(dispatchedOrdersPackagesIds[1], orders_package2.get('id'));
+});
+
+test('check allItemsDispatched returns true if all Items are dispatched otherwise returns false', function(assert){
+  assert.expect(2);
+  var store = this.store();
+  var item1 = null;
+  var item2 = null;
+  var item3 = null;
+  var unDispatcheditem = null;
+  var designation1 = null;
+  var designation2 = null;
+
+  Ember.run(function(){
+
+    store.createRecord('designation', {
+      id:               5,
+      detailType:       'StockitLocalOrder',
+      status:           'Active',
+      createdAt:        '12/07/2016',
+      updatedAt:        '12/07/2016'
+    });
+
+    store.createRecord('designation', {
+      id:               6,
+      detailType:       'StockitLocalOrder',
+      status:           'Active',
+      createdAt:        '12/07/2016',
+      updatedAt:        '12/07/2016'
+    });
+
+    store.createRecord('item', { id: 1, isDispatched: true });
```

Comment:
@viveksriram1 , is it possible that we can directly assign this to item1?
Means **item1 = store.createRecord('item', { id: 1, isDispatched: true });**, if **store.createRecord returns a model object**